### PR TITLE
Adds an example API using docker-compse with mongoDB for persistence

### PR DESCRIPTION
**Pull request by NTAWolf**
This is a working minimal example that

- serves an API
- encapsulates the current state of the code
- has a database for persistence
- uses docker-compose to do it

Hopefully this will help others who, like me, are trying to grasp how to implement microservices :-)

I'm not certain that I'm using `@hug.post` the right way - thoughts? The way I understand [the RFC](https://tools.ietf.org/html/rfc7231#section-4.3.3) is for POST requests to return 201.

### CODE REVIEW

File: examples/docker_compose_with_mongodb/app.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import os
+
+from pymongo import MongoClient
+import hug
+
+client = MongoClient('db', 27017)
```
Comment (OGKevin) [inline examples/docker_compose_with_mongodb/app.py]: There should be 2 new lines after import statements 👍

File: examples/docker_compose_with_mongodb/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (OGKevin) [inline examples/docker_compose_with_mongodb/requirements.txt]: Is this new line really needed ?

File: examples/docker_compose_with_mongodb/app.py (hunk 1 above)
Comment (NTAWolf) [reply examples/docker_compose_with_mongodb/app.py]: Thanks!

File: examples/docker_compose_with_mongodb/requirements.txt (hunk 2 above)
Comment (NTAWolf) [reply examples/docker_compose_with_mongodb/requirements.txt]: Nope, missed it. Thanks!

**Comment by timothycrosley** [comment]
Thanks for this! I think this is a great addition, and I know something many people have asked for. I think a POST is appropriately used in this context.

Thanks again!

~Timothy

**Pull request merged**